src/topology/algebra/infinite_sum.lean

```diff
@@ -202,211 +285,120 @@ mem_at_top_sets.mpr $ exists.intro fsts $ assume bs (hbs : fsts ⊆ bs),
-  (hij : ∀⦃b⦄ (h : f b ≠ 0), i (hj h) = b)
-  (hgj : ∀⦃b⦄ (h : f b ≠ 0), g (j h) = f b) :
-  (∑'i, f i) = (∑'j, g j) :=
-tsum_eq_tsum_of_has_sum_iff_has_sum $ assume a, has_sum_iff_has_sum_of_ne_zero i hi j hj hji hij hgj
-
-lemma tsum_eq_tsum_of_ne_zero_bij {f : β → α} {g : γ → α}
-  (i : Π⦃c⦄, g c ≠ 0 → β)
-  (h₁ : ∀⦃c₁ c₂⦄ (h₁ : g c₁ ≠ 0) (h₂ : g c₂ ≠ 0), i h₁ = i h₂ → c₁ = c₂)
-  (h₂ : ∀⦃b⦄, f b ≠ 0 → ∃c (h : g c ≠ 0), i h = b)
-  (h₃ : ∀⦃c⦄ (h : g c ≠ 0), f (i h) = g c) :
-  (∑'i, f i) = (∑'j, g j) :=
-tsum_eq_tsum_of_has_sum_iff_has_sum $ assume a, has_sum_iff_has_sum_of_ne_zero_bij i h₁ h₂ h₃
-
-lemma tsum_eq_tsum_of_iso (j : γ → β) (i : β → γ)
-  (h₁ : ∀x, i (j x) = x) (h₂ : ∀x, j (i x) = x) :
-  (∑'c, f (j c)) = (∑'b, f b) :=
-tsum_eq_tsum_of_has_sum_iff_has_sum $ assume a, has_sum_iff_has_sum_of_iso i h₁ h₂
-
-lemma tsum_equiv (j : γ ≃ β) : (∑'c, f (j c)) = (∑'b, f b) :=
-tsum_eq_tsum_of_iso j j.symm (by simp) (by simp)
+  (assume : summable g, (h.mpr this.has_sum).tsum_eq)
+  (assume hg : ¬ summable g,
+    have hf : ¬ summable f, from mt (e.summable_iff_of_has_sum_iff @h).1 hg,
+    by simp [tsum, hf, hg, h0])
+
+lemma tsum_eq_tsum_of_has_sum_iff_has_sum {f : β → α} {g : γ → α}
+  (h : ∀{a}, has_sum f a ↔ has_sum g a) :
+  (∑'b, f b) = (∑'c, g c) :=
+(equiv.refl α).tsum_eq_tsum_of_has_sum_iff_has_sum rfl @h
+
+lemma equiv.tsum_eq (j : γ ≃ β) (f : β → α) : (∑'c, f (j c)) = (∑'b, f b) :=
+tsum_eq_tsum_of_has_sum_iff_has_sum $ λ a, j.has_sum_iff
+
+lemma equiv.tsum_eq_tsum_of_support {f : β → α} {g : γ → α} (e : support f ≃ support g)
+  (he : ∀ x, g (e x) = f x) :
+  (∑' x, f x) = ∑' y, g y :=
+tsum_eq_tsum_of_has_sum_iff_has_sum $ λ _, e.has_sum_iff_of_support he
+
+lemma tsum_eq_tsum_of_ne_zero_bij {g : γ → α} (i : support g → β)
+  (hi : ∀ ⦃x y⦄, i x = i y → (x : γ) = y)
+  (hf : support f ⊆ set.range i) (hfg : ∀ x, f (i x) = g x) :
+  (∑' x, f x)  = ∑' y, g y :=
+tsum_eq_tsum_of_has_sum_iff_has_sum $ λ _, has_sum_iff_has_sum_of_ne_zero_bij i hi hf hfg
 
 variable [topological_add_monoid α]
 
 lemma tsum_add (hf : summable f) (hg : summable g) : (∑'b, f b + g b) = (∑'b, f b) + (∑'b, g b) :=
-tsum_eq_has_sum $ hf.has_sum.add hg.has_sum
+(hf.has_sum.add hg.has_sum).tsum_eq
 
 lemma tsum_sum {f : γ → β → α} {s : finset γ} (hf : ∀i∈s, summable (f i)) :
   (∑'b, ∑ i in s, f i b) = ∑ i in s, ∑'b, f i b :=
-tsum_eq_has_sum $ has_sum_sum $ assume i hi, (hf i hi).has_sum
+(has_sum_sum $ assume i hi, (hf i hi).has_sum).tsum_eq
 
-lemma tsum_sigma [regular_space α] {γ : β → Type*} {f : (Σb:β, γ b) → α}
+lemma tsum_sigma' [regular_space α] {γ : β → Type*} {f : (Σb:β, γ b) → α}
   (h₁ : ∀b, summable (λc, f ⟨b, c⟩)) (h₂ : summable f) : (∑'p, f p) = (∑'b c, f ⟨b, c⟩) :=
-(tsum_eq_has_sum $ h₂.has_sum.sigma (assume b, (h₁ b).has_sum)).symm
+(h₂.has_sum.sigma (assume b, (h₁ b).has_sum)).tsum_eq.symm
+
+lemma tsum_prod' [regular_space α] {f : β × γ → α} (h : summable f)
+  (h₁ : ∀b, summable (λc, f (b, c))) :
+  (∑'p, f p) = (∑'b c, f ⟨b, c⟩) :=
```

Comment:
Why do you have round brackets on the previous line and angle brackets on this line?

Reply:
Copy+paste from `sigma`.